shared-module/usb_cdc/__init__.c

```diff
@@ -259,3 +264,154 @@ bool common_hal_usb_cdc_enable(bool console, bool data) {
+    U16_TO_U8S_LE(0x0050), // wPropertyDataLength
+    // bPropertyData: “{975F44D9-0D08-43FD-8B3E-127CA8AFFF9D}”.
+    '{', 0x00, '9', 0x00, '7', 0x00, '5', 0x00, 'F', 0x00, '4', 0x00, '4', 0x00, 'D', 0x00, '9', 0x00, '-', 0x00,
+    '0', 0x00, 'D', 0x00, '0', 0x00, '8', 0x00, '-', 0x00, '4', 0x00, '3', 0x00, 'F', 0x00, 'D', 0x00, '-', 0x00,
+    '8', 0x00, 'B', 0x00, '3', 0x00, 'E', 0x00, '-', 0x00, '1', 0x00, '2', 0x00, '7', 0x00, 'C', 0x00, 'A', 0x00,
+    '8', 0x00, 'A', 0x00, 'F', 0x00, 'F', 0x00, 'F', 0x00, '9', 0x00, 'D', 0x00, '}', 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+TU_VERIFY_STATIC(sizeof(ms_os_20_descriptor_template) == MS_OS_20_DESC_LEN, "Incorrect size");
+
+static const uint8_t usb_vendor_descriptor_template[] = {
+    // Vendor Descriptor
+    0x09,        // 0 bLength
+    0x04,        // 1 bDescriptorType (Interface)
+    0xFF,        // 2 bInterfaceNumber  [SET AT RUNTIME]
+#define VENDOR_INTERFACE_INDEX 2
+    0x00,        // 3 bAlternateSetting
+    0x02,        // 4 bNumEndpoints 2
+    0xFF,        // 5 bInterfaceClass: Vendor Specific
+    0x00,        // 6 bInterfaceSubClass: NONE
+    0x00,        // 7 bInterfaceProtocol: NONE
+    0xFF,        // 8 iInterface (String Index)
+#define VENDOR_INTERFACE_STRING_INDEX 8
+
+    // Vendor OUT Endpoint Descriptor
+    0x07,        // 9 bLength
+    0x05,        // 10 bDescriptorType (Endpoint)
+    0xFF,        // 11 bEndpointAddress (IN/D2H) [SET AT RUNTIME: number]
+#define VENDOR_OUT_ENDPOINT_INDEX 11
+    0x02,        // 12 bmAttributes (Bulk)
+    #if USB_HIGHSPEED
+    0x00, 0x02,  // 13,14  wMaxPacketSize 512
+    #else
+    0x40, 0x00,  // 13,14  wMaxPacketSize 64
+    #endif
+    0x0,         // 15  bInterval 0
+
+    // Vendor IN Endpoint Descriptor
+    0x07,        // 16 bLength
+    0x05,        // 17 bDescriptorType (Endpoint)
+    0xFF,        // 18 bEndpointAddress (IN/D2H) [SET AT RUNTIME: 0x80 | number]
+#define VENDOR_IN_ENDPOINT_INDEX 18
+    0x02,        // 19 bmAttributes (Bulk)
+    0x40, 0x00,  // 20, 21 wMaxPacketSize 64
+    0x0          // 22  bInterval 0
+};
+
+static const char vendor_interface_name[] = USB_INTERFACE_NAME " WebUSB";
+
+
+bool usb_vendor_enabled(void) {
+    return usb_cdc_console_enabled();
+}
+
+size_t usb_vendor_descriptor_length(void) {
+    return sizeof(usb_vendor_descriptor_template);
+}
+
+static supervisor_allocation *ms_os_20_descriptor_allocation;
+
+size_t vendor_ms_os_20_descriptor_length() {
+    return sizeof(ms_os_20_descriptor_template);
+}
+uint8_t const *vendor_ms_os_20_descriptor() {
+    return (uint8_t *)ms_os_20_descriptor_allocation->ptr;
+}
+
+
+size_t usb_vendor_add_descriptor(uint8_t *descriptor_buf, descriptor_counts_t *descriptor_counts, uint8_t *current_interface_string) {
+
+    if (ms_os_20_descriptor_template[MS_OS_20_ITF_NUM_OFFSET] == MS_OS_20_ITF_NUM_MAGIC) {
+        ms_os_20_descriptor_allocation =
+            allocate_memory(align32_size(sizeof(ms_os_20_descriptor_template)),
```

Comment:
I think all new sites of `allocate_memory` need to be accounted for in the macro CIRCUITPY_SUPERVISOR_IMMOVABLE_ALLOC_COUNT or CIRCUITPY_SUPERVISOR_MOVABLE_ALLOC_COUNT in `supervisor/shared/memory.c`

Reply:
Thanks for that. I'll add to the enum.   I won't push until we have a clearer idea of what we're doing with the WebUSB support, since the CI build is quite expensive. :-)

Reply:
Changes pushed to track the non-movable memory.